mpi-proxy-split/mpi_plugin.cpp

```diff
@@ -361,16 +362,28 @@ computeUnionOfCkptImageAddresses()
   // Adjust libsStart to make 4GB space.
   libsStart = (void *)((uint64_t)libsStart - 4 * ONEGB);
 
-  string kvdb = "MANA_CKPT_UNION";
-  dmtcp_kvdb64(DMTCP_KVDB_MIN, kvdb.c_str(), 0, (int64_t) libsStart);
-  dmtcp_kvdb64(DMTCP_KVDB_MAX, kvdb.c_str(), 1, (int64_t) libsEnd);
-  dmtcp_kvdb64(DMTCP_KVDB_MIN, kvdb.c_str(), 2, (int64_t) highMemStart);
+  string workerPath("/worker/" + string(dmtcp_get_uniquepid_str()));
+  string libsStartStr = jalib::XToString(libsStart);
+  string libsEndStr = jalib::XToString(libsEnd);
+  string highMemStartStr = jalib::XToString(highMemStart);
 
-  dmtcp_global_barrier(kvdb.c_str());
+  dmtcp::kvdb::set(workerPath, "MANA_libsStart", libsStartStr);
+  dmtcp::kvdb::set(workerPath, "MANA_libsEnd", libsEndStr);
+  dmtcp::kvdb::set(workerPath, "MANA_highMemStart", highMemStartStr);
 
-  JASSERT(dmtcp_kvdb64_get(kvdb.c_str(), 0, (int64_t*) &minLibsStart) == 0);
-  JASSERT(dmtcp_kvdb64_get(kvdb.c_str(), 1, (int64_t*) &maxLibsEnd) == 0);
-  JASSERT(dmtcp_kvdb64_get(kvdb.c_str(), 2, (int64_t*) &minHighMemStart) == 0);
+  string kvdb = "/plugin/MANA/CKPT_UNION";
+  dmtcp::kvdb::request64(dmtcp::kvdb::KVDBRequest::MIN, kvdb, "libsStart", (int64_t) libsStart);
```

Comment:
@karya0 could you remind me if we need to check the return value from request64 or not?

Reply:
Good catch. Added an assert.